src/osd/PrimaryLogPG.cc

```diff
@@ -8313,6 +8327,26 @@ int PrimaryLogPG::_rollback_to(OpContext *ctx, OSDOp& op)
     block_write_on_degraded_snap(missing_oid, ctx->op);
     return ret;
   }
+  /*
+   * In rollback, if the head object is not manfest and the rollback_to is manifest,
+   * the head object will become the manifest object. At this point,
+   * we need to check adjacent clones beside the head object to calculate 
+   * correct reference count for deduped chunks because the head object is now 
+   * manifest. The reverse is also true---the head object is manifest, but the rollback_to
+   * is not manifest.
+   * Therefore, the following lines inserts the op to the waiting queue to wait until
+   * unreadable object is recovered if either adjacent clones is 
+   * unreadable to calculate chunk references.
+   */
+  if ((oi.has_manifest() && oi.manifest.is_chunked()) ||
+      (rollback_to->obs.oi.has_manifest() &&
+      rollback_to->obs.oi.manifest.is_chunked())) {
+    snapid_t id = do_recover_adjacent_clones(ctx->obc, ctx->op);
+    if (id != snapid_t()) {
+      objects_blocked_on_unreadable_adjacent_snap[soid.get_head()] = id;
```

Comment:
Let's introduce a helper block_write_on_unreadable_snap to match the above structure.

Reply:
Done.